nixos/modules/services/networking/mbim-network.nix

```diff
@@ -0,0 +1,126 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mbim-network;
+
+  # Transforms "/dev/cdc-wdm0" into "mbim-network-dev-cdc-wdm0" for file and
+  # systemd unit names.
+  cleanName = name: "mbim-network${replaceStrings [ "/" ] [ "-" ] name}";
```

Comment:
maybe you can use escapeSystemdPath in nixos/lib/utils instead ?

Reply:
I gave this a try but it seems like the existing hyphen in the device name trips it up:

```
  # Transforms "/dev/cdc-wdm0" into "mbim-network-dev-cdc-wdm0" for file and
  # systemd unit names.
  cleanName = name: "mbim-network-${utils.escapeSystemdPath name}";
```
```
[matt@routnerr-2:/etc/nixos/lib/modules]$ sudo nixos-rebuild -j0 switch
building Nix...
building the system configuration...
error: The path name 'mbim-network-dev-cdc\x2dwdm0.conf' is invalid: the '\' character is invalid. Path names are alphanumeric and can include the symbols +-._?= and must not begin with a period. Note: If 'mbim-network-dev-cdc\x2dwdm0.conf' is a source file and you cannot rename it on disk, builtins.path { name = ... } can be used to give it an alternative name.
(use '--show-trace' to show detailed location information)

```
```

```